### PR TITLE
net-misc/r8125: fixed PTP and migrated to linux-mod-r1

### PR DESCRIPTION
**Pull request by Karlson2k**
Together with migration to linux-mod-r1.eclass I also added more predefined Makefile variables to avoid extra detection by Makefile and to make sure that Makefile will use precisely the same values as expected by portage. This should also fix ability to build for other than currently loaded kernel.

Unfortunately linux-mod-r1.eclass does not document available variables and helpers (except standard stages functions).
It would be nice to have some helpers to extract module build parameters, like "install dir", "source dir". Currently it is solved in these .ebuilds by additional local variables to avoid strings duplication and simplify maintainability. 
However, I'd repeat, helper functions or official variables would be very helpful here.

### CODE REVIEW

**Comment by Karlson2k** [comment]
To simplify the review of migration to linux-mod-r1.eclass.
Here are the variables used by the Makefile:
https://github.com/awesometic/realtek-r8125-dkms/blob/27161e7c098631ad960bc6cc6f48ad131baac8d3/src/Makefile#L135

The directory for the module is expected to be `kernel/drivers/net/ethernet/realtek`, see https://github.com/awesometic/realtek-r8125-dkms/blob/27161e7c098631ad960bc6cc6f48ad131baac8d3/src/Makefile#L138

**Comment by Karlson2k** [comment]
@mpagano @thesamesam @juippis 
Could you please check this PR?

**Comment by thesamesam** [comment]
@ionenwks Would you mind taking a look? I'm just wondering if the BASEDIR/DRIVERDIR bit can be simplified mainly.

File: net-misc/r8125/r8125-9.011.01-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+EAPI=8
+
+inherit linux-mod-r1
+
+DESCRIPTION="r8125 vendor driver for Realtek RTL8125 PCI-E NICs"
+HOMEPAGE="https://www.realtek.com/en/component/zoo/category/network-interface-controllers-10-100-1000m-gigabit-ethernet-pci-express-software"
+
+SRC_URI="http://rtitwww.realtek.com/rtdrivers/cn/nic1/${P}.tar.bz2"
+
+LICENSE="GPL-2"
```
Comment (ionenwks) [inline net-misc/r8125/r8125-9.011.01-r1.ebuild:13]: Happened to loo at a .c file and seen there's a "any later version"
```suggestion
LICENSE="GPL-2+"
```

**Review by ionenwks — COMMENTED** [review]
>Unfortunately linux-mod-r1.eclass does not document available variables and helpers (except standard stages functions).

That's because the variables are from linux-info.eclass, that's where you should look for documentation of these. linux-mod-r1 just guarantee that linux-info's variables will be initialized after pkg_setup.

>It would be nice to have some helpers to extract module build parameters, like "install dir", "source dir". Currently it is solved in these .ebuilds by additional local variables to avoid strings duplication and simplify maintainability.

All you normally need is figure out which variable the Makefile using for kernel dir (this varies, so the eclass can't know -- each module just does what they want). Then pass  KV_OUT_DIR from linux-info.

File: net-misc/r8125/r8125-9.011.01-r1.ebuild (hunk 1 above)
Comment (Karlson2k) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild:13]: There are conflicting terms in each and every file.
The first line is `SPDX-License-Identifier: GPL-2.0-only` and below in the same file `either version 2 of the License, or (at your option) any later version.`
For example, see:
* https://github.com/awesometic/realtek-r8125-dkms/blob/9.011.00-1/src/r8125_n.c#L1
* https://github.com/awesometic/realtek-r8125-dkms/blob/9.011.00-1/src/r8125_n.c#L12C24-L13C21

As `GPL-2.0+` sources could be used under `GPL-2.0-only` terms, I'd leave `GPL-2.0-only` which fits unambiguously.

**Comment by Karlson2k** [comment]
@ionenwks 
I've updated the PR.
All extra variables were removed. 
Makefile now uses its own detection which slow down build a little, but ebuild is now minimal and simple.

File: net-misc/r8125/r8125-9.011.01-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	local modargs=(
+		# Build parameters
+		KERNELDIR="${KV_OUT_DIR}"
+		DRIVERDIR="/lib/modules/${KV_FULL}/${inst_subdir}"
+		PWD="$S/${modsrc_subdir}"
+		RKTDIR="${inst_dir}"
+		BASEDIR="/lib/modules/${KV_FULL}"
+		KVER="${KV_FULL}"
+		KMAJ="${KV_MAJOR}" KMIN="${KV_MINOR}" KREV="${KV_PATCH}"
+		KERNEL_GCC_VERSION="" # Not actually used by Makefile
```
[the remark this answers is not in the document]
Comment (Karlson2k) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild]: OK.
Removed extra vars and simplified.

File: net-misc/r8125/r8125-9.011.01-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	"${FILESDIR}/${P}-linux-6.2.patch" # bug 908645
+)
+
+CONFIG_CHECK="~!R8169"
+WARNING_R8169="CONFIG_R8169 is enabled. ${PN} will not be loaded unless kernel driver Realtek 8169 PCI Gigabit Ethernet (CONFIG_R8169) is DISABLED."
+
+src_compile() {
+	local inst_subdir='kernel/drivers/net/ethernet/realtek'
+	local modsrc_subdir='src'
+	local modlist=( ${PN}=${inst_subdir}:${modsrc_subdir}::"print_vars modules" )
```
[the remark this answers is not in the document]
Comment (Karlson2k) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild]: OK.
Simplfied.

File: net-misc/r8125/r8125-9.011.01-r1.ebuild (hunk 1 above)
Comment (ionenwks) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild]: I see, when there's both I typically just list both rather than combine.

Comment (Karlson2k) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild]: The union of `GPL-2.0-only` and `GPL-2.0+` is `GPL-2.0-only`. :)
Listing both terms as both applicable at the same time (not just one of them) is the same as the `-only`.

Comment (Karlson2k) [reply net-misc/r8125/r8125-9.011.01-r1.ebuild]: Maybe we should ignore "SPDX identifier" as the full licence wording in the same file has the full force, while "SPDX identified" does not enforce anything legally?

**Comment by ionenwks** [comment]
> I've updated the PR. All extra variables were removed. Makefile now uses its own detection which slow down build a little, but ebuild is now minimal and simple.

I'd be mildly concerned if the find calls crawled the full /lib/modules (can be lot of files), but it checks in a rather specific directory and the rest is just cheap cat/sed/uname. So personally wouldn't worry about it.

**Comment by Karlson2k** [comment]
The `find` is called two time:
https://github.com/awesometic/realtek-r8125-dkms/blob/9.011.00-1/src/Makefile#L138
Unless something heavy in this directory, it should be quick,

Alternatively, we may simply use empty assignment for this (and some other) variables.

**Comment by Karlson2k** [comment]
@ionenwks The PR is ready for review. :)

**Comment by ionenwks** [comment]
Ah sorry, I didn't have anything to add and thought sam was going to have a last look + merge -- so thought I was done here.

Guess I'll test the other ebuilds quick too and merge it myself then.

**Pull request closed without merging**